### PR TITLE
Add nodejs terraform template

### PR DESCRIPTION
**Pull request by HadwaAbdelhalem**
This PR adds template-nodejs-mongo terraform iac code by  

-  Create new Azure-Samples repo

-  Build out required terraform infra components

Addresses #231

### CODE REVIEW

**Review by ellismg — APPROVED** [review]

File: templates/todo/projects/nodejs-mongo/.repo/bicep/repo.yaml [hunk 1]
```diff
@@ -31,6 +31,11 @@ repo:
         patterns:
           - "**/azure.@(yml|yaml)"
 
+      - from: "PLACEHOLDERIACTOOLS"
```
Comment (ellismg) [inline templates/todo/projects/nodejs-mongo/.repo/bicep/repo.yaml:34]: This is a cute trick!  Love how we were able to share so much between the README's with this.

Comment (wbreza) [reply templates/todo/projects/nodejs-mongo/.repo/bicep/repo.yaml:34]: Yes, loving the unintended usage of the repoman rewrite rules 🤣

**Review by wbreza — APPROVED** [review]
Looks great.  Opportunity to share a lot of the modules in the future - but this is good enough for now.

File: templates/todo/projects/nodejs-mongo/.repo/terraform/azure.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/v1.0/azure.yaml.json
+
+name: todo-nodejs-mongo-terraform
```
Comment (vhvb1989) [inline templates/todo/projects/nodejs-mongo/.repo/terraform/azure.yaml:3]: Add an entry in `cli/azd/resources/templates.json` for this template to be listed on `azd init` and `azd templates`

File: templates/todo/projects/nodejs-mongo/.repo/terraform/azure.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/v1.0/azure.yaml.json
+
+name: todo-nodejs-mongo-terraform
+metadata:
+  template: todo-nodejs-mongo-terraform@0.0.1-beta
```
Comment (vhvb1989) [inline templates/todo/projects/nodejs-mongo/.repo/terraform/azure.yaml:5]: Add an entry to Changelog about a new template was added?

**Review by vhvb1989 — COMMENTED** [review]
Suggesting to integrate with azd a little more by adding entries

**Review by hattan — APPROVED** [review]
LGTM

File: templates/todo/projects/nodejs-mongo/.repo/terraform/azure.yaml (hunk 3 above)
Comment (HadwaAbdelhalem) [reply templates/todo/projects/nodejs-mongo/.repo/terraform/azure.yaml:5]: added

File: templates/todo/projects/nodejs-mongo/.repo/terraform/azure.yaml (hunk 2 above)
Comment (HadwaAbdelhalem) [reply templates/todo/projects/nodejs-mongo/.repo/terraform/azure.yaml:3]: added

File: templates/todo/projects/nodejs-mongo/.repo/terraform/azure.yaml (hunk 3 above)
Comment (HadwaAbdelhalem) [reply templates/todo/projects/nodejs-mongo/.repo/terraform/azure.yaml:5]: not sure what the standard messages are. will resolve to merge now and update it later

**Pull request merged**